tests/unit/lib/utils/test_hash.py

```diff
@@ -13,6 +14,42 @@ def setUp(self):
     def tearDown(self):
         shutil.rmtree(self.temp_dir, ignore_errors=True)
 
+    def test_dir_hash_independent_of_location(self):
+        temp_dir1 = os.path.join(self.temp_dir, "temp-dir-1")
+        os.mkdir(temp_dir1)
+        with open(os.path.join(temp_dir1, "test-file"), "w+") as f:
+            f.write("Testfile")
+        checksum1 = dir_checksum(temp_dir1)
+
+        temp_dir2 = shutil.move(temp_dir1, os.path.join(self.temp_dir, "temp-dir-2"))
+        checksum2 = dir_checksum(temp_dir2)
+
+        self.assertEqual(checksum1, checksum2)
+
+    def test_dir_hash_independent_of_file_order(self):
+        file1 = tempfile.NamedTemporaryFile(delete=False, dir=self.temp_dir)
```

Comment:
small nit: can we name the files ourselves? this way we can showcase the sorting to work.

Because otherwise its non-determinstic, the names of the files will be different every run.

Reply:
we pass it both ways, and the name of the files can never be the same, so I rescind my nit. Approving.